mlflow/spark.py

```diff
@@ -208,33 +215,26 @@ def log_model(
         spark_model = PipelineModel([spark_model])
     run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
     run_root_artifact_uri = mlflow.get_artifact_uri()
+    if _should_use_mlflowdbfs(run_root_artifact_uri):
+        mlflowdbfs_path = _mlflowdbfs_path(run_id, artifact_path)
+        with databricks_utils.MlflowCredentialContext(
+            get_databricks_profile_uri_from_artifact_uri(run_root_artifact_uri)
+        ):
+            try:
+                spark_model.save(mlflowdbfs_path)
+                return
```

Comment:
We still need to log the MLmodel file, conda.yaml file, etc. Otherwise, the spark model cannot be reloaded with `mlflow.spark.load_model()`. I'm seeing the following error on Databricks:

```
Failed to download an "MLmodel" model file from "dbfs:/databricks/mlflow-tracking/4379665129519901/6675a7f8ebdd4af19cb9054e9edf4e71/artifacts/foobaz4": The following failures occurred while downloading one or more artifacts from dbfs:/databricks/mlflow-tracking/4379665129519901/6675a7f8ebdd4af19cb9054e9edf4e71/artifacts/foobaz4: {'MLmodel': 'MlflowException(\'API request failed with exception 404 Client Error: Not Found for url:...
```

Reply:
I tried to log a model with mlflowdbfs disabled and then load it with mlflowdbfs *enabled*; I hit the following error:

```
import os
os.environ["DISABLE_MLFLOWDBFS"] = "true"
mlflow.spark.log_model(pipeline_model, "foobaz")
os.environ["DISABLE_MLFLOWDBFS"] = ""
mlflow.spark.load_model("runs:/6675a7f8ebdd4af19cb9054e9edf4e71/foobaz")
```

```
java.io.FileNotFoundException: This path does not exist in the artifact repository! Path: artifacts?run_id=6675a7f8ebdd4af19cb9054e9edf4e71&path=/sparkml/metadata
```

Even though this path does seem to exist. Can we investigate?